code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/TokenUtils.java

```diff
@@ -284,4 +300,55 @@ public Account getOrCreateAccount(Identity gotIdentity, Set<Identity> identities
         }
         return account;
     }
+
+
+    public Token createToken(Set<Identity> identities, Account account) {
```

Comment:
This logic was moved from 3 different locations where the token was created in almost the exact same manner. Now the token is created in the same manner, while allowing any modification that is needed to the associated account to be done before the token is returned but only after the user has been authenticated.